aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1042 @@
+            let dora_public_key = vector::pop_back(&mut dora_public_key);
+            let network_public_key = vector::pop_back(&mut network_public_key);
+            let elgamal_pub_key = vector::pop_back(&mut elgamal_pub_key);
+            let network_port = vector::pop_back(&mut network_port);
+            let rpc_port = vector::pop_back(&mut rpc_port);
+            let dora_node = DoraNodeInfo {
+                ip_public_address: copy ip_public_address,
+                dora_public_key: copy dora_public_key,
+                network_public_key: copy network_public_key,
+                elgamal_pub_key: copy elgamal_pub_key,
+                network_port: network_port,
+                rpc_port: rpc_port,
+            };
+            vector::push_back(&mut dora_node_info, dora_node);
+            // Also update the node_to_committee_map
+            let node_address = vector::pop_back(&mut node_addresses_for_iteration);
+            simple_map::upsert(&mut committee_store.node_to_committee_map, node_address, id);
+        };
+        let committee_info = CommitteeInfo {
+            map: simple_map::new_from(node_addresses, dora_node_info),
+            has_valid_dkg: false,
+        };
+        let event_handler = borrow_global_mut<SupraCommitteeEventHandler>(owner_address);
+        if (!simple_map::contains_key(&committee_store.committee_map, &id)) {
+            emit_event(
+                &mut event_handler.add_committee,
+                AddCommitteeEvent {
+                    committee_id: id,
+                    committee_info: copy committee_info
+                }, )
+        } else {
+            let old_committee_info = *simple_map::borrow(&committee_store.committee_map, &id);
+            emit_event(
+                &mut event_handler.update_committee,
+                UpdateCommitteeEvent {
+                    committee_id: id,
+                    old_committee_info: old_committee_info,
+                    new_committee_info: committee_info
+                },
+            )
+        };
+        simple_map::upsert(&mut committee_store.committee_map, id, committee_info);
+    }
+
+    /// Add the committee in bulk
+    public entry fun upsert_committee_bulk(
+        com_store_addr: address,
+        owner_signer: &signer,
+        ids: vector<u64>,
+        node_addresses_bulk: vector<vector<address>>,
+        ip_public_address_bulk: vector<vector<vector<u8>>>,
+        dora_public_key_bulk: vector<vector<vector<u8>>>,
+        network_public_key_bulk: vector<vector<vector<u8>>>,
+        elgamal_pub_key_bulk: vector<vector<vector<u8>>>,
+        network_port_bulk: vector<vector<u16>>,
+        rpc_por_bulkt: vector<vector<u16>>,
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Assert the length of the vector for two are the same
+        assert!(
+            vector::length(&ids) == vector::length(&node_addresses_bulk),
```

Comment:
Same comment as I mentioned above regarding length,

Also, you haven't checked the Owner's Capability here I think it's because later on the `upsert_committee` function did the same.

Reply:
@nizam-supraoracles Thanks, addressed in new commit. 
>Also, you haven't checked the Owner's Capability here I think it's because later on the upsert_committee function did the same.
Yes, because in bulk operation, it internally calls single upsert operation, which does the check